src/charset.c

```diff
@@ -527,26 +527,31 @@ transchar_buf(buf_T *buf, int c)
  */
     char_u *
 transchar_byte(int c)
+{
+    return transchar_byte_buf(curbuf, c);
+}
+
+    char_u *
+transchar_byte_buf(buf_T *buf,int c)
 {
     if (enc_utf8 && c >= 0x80)
     {
-	transchar_nonprint(curbuf, transchar_charbuf, c);
-	return transchar_charbuf;
+        transchar_nonprint(buf, transchar_charbuf, c);
+        return transchar_charbuf;
```

Comment:
```suggestion
	transchar_nonprint(buf, transchar_charbuf, c);
	return transchar_charbuf;
```
Tab characters should be used here.